folsom/src/main/java/com/spotify/folsom/client/Request.java

```diff
@@ -28,9 +28,11 @@
 
 public abstract class Request<V> extends CompletableFuture<V> {
   protected final byte[] key;
+  protected final long createdNanos;
```

Comment:
Why moving this timestamp to the `Request` instead of keeping it in `TimeoutChecker` ?